addons/binding/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/ApplianceGroheOndusHandler.java

```diff
@@ -0,0 +1,247 @@
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.grohe.ondus.api.OndusService;
+import org.grohe.ondus.api.model.Appliance;
+import org.grohe.ondus.api.model.ApplianceCommand;
+import org.grohe.ondus.api.model.ApplianceData;
+import org.grohe.ondus.api.model.ApplianceData.Measurement;
+import org.grohe.ondus.api.model.Location;
+import org.grohe.ondus.api.model.Room;
+import org.openhab.binding.groheondus.internal.ApplianceGroheOndusConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Florian Schmidt and Arne Wohlert - Initial contribution
+ */
+@NonNullByDefault
+public class ApplianceGroheOndusHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ApplianceGroheOndusHandler.class);
+
+    private @Nullable ApplianceGroheOndusConfiguration config;
+
+    public ApplianceGroheOndusHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(ApplianceGroheOndusConfiguration.class);
+        scheduler.submit(() -> {
+            updateChannels();
+            updateStatus(ThingStatus.ONLINE);
+        });
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        super.channelLinked(channelUID);
+
+        OndusService ondusService = getOndusService();
+        if (ondusService == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    "No initialized OndusService available from bridge.");
+            return;
+        }
+        Appliance appliance = getAppliance(ondusService);
+        if (appliance == null) {
+            return;
+        }
+        updateChannel(channelUID, appliance, getLastMeasurement(appliance));
+    }
+
+    public void updateChannels() {
+        OndusService ondusService = getOndusService();
+        if (ondusService == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    "No initialized OndusService available from bridge.");
+            return;
+        }
+        Appliance appliance = getAppliance(ondusService);
+        if (appliance == null) {
+            return;
+        }
+
+        Measurement measurement = getLastMeasurement(appliance);
+        getThing().getChannels().forEach(channel -> updateChannel(channel.getUID(), appliance, measurement));
+
+        int pollingInterval = getPollingInterval(appliance);
+        scheduler.schedule(this::updateChannels, pollingInterval, TimeUnit.SECONDS);
```

Comment:
This looks bad. You are recursively scheduling new jobs in the method that is called by the scheduled jobs and you do not keep track of them and never ever cancel them again?

Reply:
This schedule call is a one-shot only, so there's only one scheduled job at a time, which is executed after the giving amount of time (pollingInterval). This ensures, that the job is started with the current pollingInterval, e.g. is it changed in the API or configuration.

There's one down side here: When the update of the channel fails because of an exception, the scheduler is not re-scheduled for the next one-shot :/

Reply:
Ah, your are right sorry, I didn't see that it isn't a repeating scheduling here.
Nonetheless, I don't think it is a good practice to schedule it from within a job. I'd prefer if you'd schedule it as a recurring job, that's how all other bindings do it as well. The benefit is also that you have a handle to it and can cancel the job in the dispose method, which you are currently not doing (so you might execute this code even seconds after handler disposal).
If the polling frequency configuration is changed, the handler is anyhow restarted, so it would automatically do the correct schedule in that case as well.